src/Microsoft.VisualStudio.ProjectSystem.Managed/StringComparers.cs

```diff
@@ -13,16 +15,31 @@ namespace Microsoft.VisualStudio
     /// </remarks>
     internal static class StringComparers
     {
-        public static IEqualityComparer<string?> WorkspaceProjectContextIds => StringComparer.Ordinal;
-        public static IEqualityComparer<string> Paths => StringComparer.OrdinalIgnoreCase;
-        public static IEqualityComparer<string> PropertyNames => StringComparer.OrdinalIgnoreCase;
-        public static IEqualityComparer<string> PropertyLiteralValues => StringComparer.OrdinalIgnoreCase;
-        public static IEqualityComparer<string> PropertyValues => StringComparer.Ordinal;
-        public static IEqualityComparer<string> RuleNames => StringComparer.OrdinalIgnoreCase;
-        public static IEqualityComparer<string> ConfigurationDimensionNames => StringComparer.Ordinal;
-        public static IEqualityComparer<string> DependencyProviderTypes => StringComparer.OrdinalIgnoreCase;
-        public static IEqualityComparer<string> ItemNames => StringComparer.OrdinalIgnoreCase;
-        public static IEqualityComparer<string> ItemTypes => StringComparer.OrdinalIgnoreCase;
+        public static StringComparer WorkspaceProjectContextIds => StringComparer.Ordinal;
```

Comment:
These need to be typed as "StringComparer" as we use both `IEqualityComparer<string>` and `IComparer<string>`.